src/module/commenthandler.js

```diff
@@ -0,0 +1,64 @@
+const movieAPI = 'https://api.tvmaze.com/shows/1/episodes?specials=1';
+const commentAPI = 'https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/apps/Bz8sde1lr8WmKzCrHh97/comments';
+const likesAPI = 'https://us-central1-involvement-api.cloudfunctions.net/capstoneApi/apps/hHoxKTRKINp4PN8g78ys/likes';
+
+const fetchData = async (url, options = {}) => {
+  try {
+    const response = await fetch(url, options);
+    if (!response.ok) throw new Error(`Error: ${response.status} ${response.statusText}`);
+    return await response.json();
+  } catch (error) {
+    return error;
+  }
+};
+
+const fetchComments = async (itemId) => {
+  try {
+    const data = await fetchData(`${commentAPI}?item_id=${itemId}`);
+    const commentList = document.querySelector('.comment-list');
+    commentList.innerHTML = '';
+    data.forEach((item) => {
+      const commentEl = document.createElement('li');
+      commentEl.classList.add('comment-item');
+      commentEl.innerHTML = `<h3>${item.creation_date} ${item.username}: ${item.comment}</h3>`;
+      commentList.appendChild(commentEl);
+    });
+    document.querySelector('.comments-title').innerHTML = `Comments (${data.length})`;
+    return data;
+  } catch (error) {
+    return error;
+  }
+};
+
+const addComment = async (id, userName, comment) => {
+  try {
+    const response = await fetchData(commentAPI, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({
+        item_id: id,
+        username: userName,
+        comment,
+      }),
+    });
+    if (!response.ok) throw new Error(`Error creating comment for id ${id}`);
+    return response;
+  } catch (error) {
+    return error;
+  }
+};
```

Comment:
- [ ] Great job on updating the code! However, it looks like there might be a new bug that needs to be fixed. Before the update, users were able to successfully add new comments. Now, there seems to be an issue where the item_id becomes undefined, preventing users from adding comments with the movie's ID. It would be great if you could take a look and address this error. Thank you! 👋 

[Capston-Kanban-projects.webm](https://github.com/richie1988/KanbanCapstone/assets/55840999/3b8c100e-09b1-4b85-9981-fa2410240160)